pkg/infrastructure/gcp/clusterapi/clusterapi.go

```diff
@@ -261,15 +261,15 @@ func (p Provider) DestroyBootstrap(ctx context.Context, in clusterapi.BootstrapD
 	projectID := in.Metadata.GCP.ProjectID
 	if in.Metadata.GCP.NetworkProjectID != "" {
 		projectID = in.Metadata.GCP.NetworkProjectID
-
-		createFwRules, err := icgcp.HasPermissions(ctx, projectID, []string{icgcp.DeleteGCPFirewallPermission}, in.Metadata.GCP.ServiceEndpoints)
-		if err != nil {
-			return fmt.Errorf("failed to remove bootstrap firewall rules: %w", err)
-		}
-		if !createFwRules {
-			return nil
-		}
 	}
+	createFwRules, err := icgcp.HasPermissions(ctx, projectID, []string{icgcp.DeleteGCPFirewallPermission}, in.Metadata.GCP.ServiceEndpoints)
```

Comment:
nit: We should use `deleteFwRules, err := ...` as we are checking delete permission?